newIDE/app/public/external/utils/path-editor.js

```diff
@@ -15,25 +16,30 @@ export const createPathEditorHeader = ({
   extension,
   headerStyle,
 }) => {
-  if (fs.lstatSync(initialResourcePath).isDirectory()) {
-    initialResourcePath = initialResourcePath + '/NewFile' + extension;
+  if (fs.existsSync(initialResourcePath)) {
+    if (fs.lstatSync(initialResourcePath).isDirectory()) {
+      initialResourcePath = initialResourcePath + '\\NewFile' + extension;
+    }
+  } else {
+    initialResourcePath = projectPath + '\\NewFile' + extension;
   }
 
+  initialResourcePath = path.normalize(initialResourcePath)
   const headerObject = {
     state: {
       folderPath: initialResourcePath.substring(
         0,
-        initialResourcePath.lastIndexOf('/')
+        initialResourcePath.lastIndexOf('\\')
```

Comment:
Totally missed that. This won't work on Linux and macOS. Instead you should use `path` methods rather than doing path manipulations by yourself.
Get the folder name with: https://nodejs.org/api/path.html#path_path_dirname_path
Extension name with: https://nodejs.org/api/path.html#path_path_extname_path
name with: https://nodejs.org/api/path.html#path_path_basename_path_ext

(in your case you may need to do: `name: path.basename(initialResourcePath, path.extname(initialResourcePath))` as you want the name without the extension (read carefully basename doc and the rest of the doc to be sure that it's doing what you want).

As a rule of thumb, never do path manipulation by yourself. It's error prone and actually more difficult that what  one can think.

Reply:
that is very good advice. Thank you @4ian :) I have now updated it and hope it works better on mac os.
Come to think of it, I did have some headache dealing with this path business on an app I developed with godot engine ages ago- most probably because I used this same string manipulation instead of using the path methods.
This makes me wonder now if godot has path methods